src/services/TaxFormProcessorService.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * This service processes the events fired when users agree to Standard Topcoder terms and NDA terms
+ */
+
+const Joi = require('@hapi/joi')
+const _ = require('lodash')
+const logger = require('../common/logger')
+const helper = require('../common/helper')
+const constants = require('../common/constants')
+
+/**
+  * Process the terms agreement event message
+  * @param {Object} message the kafka message
+  */
+async function processMessage (message) {
+  logger.info(`Process message of taxForm ${message.payload.taxForm}, userId ${message.payload.userId}, handle ${message.payload.Handle}`)
+  // Get the user handle from members api
+  const handle = _.get(message.payload, 'Handle')
+
+  // Get the member Onboarding Checklist traits
+  const onboardingChecklistTraits = await helper.getMemberTraits(handle, constants.ONBOARDING_CHECKLIST_TRAIT_ID)
+
+  // Initialize the terms traits data object
+  const termsTraitsData = {
+    status: constants.CHECKLIST_STATUS.COMPLETED,
+    message: constants.CHECKLIST_MESSAGE.SUCCESS,
+    date: new Date().getTime()
```

Comment:
@xxcxy can you please add the tax form name as well here? or better change the field name for the trait to be `w9_tax_form_submitted` and `w8ben_tax_form_submitted` instead of `tax_form_submitted`. Lets map the tax form name to these keys in our code instead of dynamically driving them from the form name i.e. set the field name oto `w9_tax_form_submitted` when taxForm starts with `W-9` and set it to `w8ben_tax_form_submitted` when taxForm starts with `W-8BEN`.

Reply:
done.

Reply:
Thanks